src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,123 @@
+
+    /**
+     * Prints welcome message.
+     */
+    public void showWelcome() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Prints line.
+     */
+    public void showLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Reads user command.
+     *
+     * @return User's command.
+     */
+    public String readCommand() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Prints loading error message.
+     */
+    public void showLoadingError() {
+        System.out.println("There is some error in the file");
+    }
+
+    /**
+     * Shows error message.
+     *
+     * @param errorMessage Error message.
+     */
+    public void showError(String errorMessage) {
+        System.out.println(errorMessage);
+    }
+
+    /**
+     * Prints exit message.
+     */
+    public void showLeaving() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Prints tasks.
+     *
+     * @param tasks Tasks to be printed.
+     */
+    public void listTasks(TaskList tasks) {
+        ArrayList<Task> list = tasks.getList();
+        for (int i = 0; i < list.size(); i++) {
+            String todo = String.format("%d. %s", i + 1, list.get(i).toString());
+            System.out.println(todo);
+        }
+    }
+
+    /**
+     * Prints done message.
+     *
+     * @param task The task that is done.
+     */
+    public void showDone(Task task) {
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.print("  ");
+        System.out.println(task.toString());
+    }
+
+    /**
+     * Prints delete message.
+     *
+     * @param task The task be deleted.
+     * @param listSize The size of the tasklist.
+     */
+    public void showDelete(Task task, int listSize) {
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(String.format("  %s", task.toString()));
+        System.out.println(String.format("Now you have %d tasks in the list.", listSize));
+    }
+
+    /**
+     * Prints add message.
+     *
+     * @param task The task to be added.
+     * @param listSize The size of the tasklist.
+     */
+    public void showAdd(Task task, int listSize) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(String.format("  %s", task.toString()));
+        System.out.println(String.format("Now you have %d tasks in the list.", listSize));
+    }
+
+    /**
+     * Prints find message.
+     *
+     * @param tasklist The tasks contains keyword.
+     */
+    public void showFind(ArrayList<String> tasklist) {
```

Comment:
For the part to print out the result list (showFind() and listTasks()), you may want to consider abstract away the printing part for the list and put it in one method to prevent repetition. You can then call the new methbod in showFind() and listTasks() to help you print the result list only (without the top description like "Here are your matching...").

e.g. 
```
public void showFind(ArrayList<String> tasklist) {
    System.out.println("Here are the matching tasks in your list:");
    printList(tasklist);
}
```